### DISCUSSION TITLE
请问各位大佬如何中转机场的订阅信息（流量使用情况）？

### DISCUSSION THREAD

**Discussion by refined-fish** [vernesong/OpenClash#4116] [Q&A] [answered]
## 环境描述
本人有多台openWRT设备使用同样的机场订阅，但机场的订阅链接偶尔会变更。每次变更都需要手动配置给每一台设备，非常不便。因此我利用其中一台“机器A”，由机器A向机场拉取配置文件，再通过uhttpd对外提供配置文件的下载服务。其余机器的订阅地址填写机器A的uhttpd服务器地址，实现配置文件的中转分发。这样每次只需要维护机器A的订阅链接，还能增强其余机器的订阅文件拉取速度和稳定性。

## 问题
由于经过了简易的中转，订阅信息丢失，现无法显示流量使用情况。经过简单了解得知openclash似乎会通过某个api向订阅地址请求流量使用情况的json信息，想求助各位这个api具体是什么，又是否有简单的将此请求也中转，或者其他能够恢复订阅信息的办法。
![image](https://github.com/user-attachments/assets/8e29cc93-ef2a-4414-9e4f-52f7a40d2079)

**Comment by fejeIaqbhbv** [comment]
没api，这玩意就藏在订阅链接里
源代码形式打开订阅链接就行了
比如这是我的
`subscription-userinfo: upload=5087212927; download=66988282145; total=536870912000; expire=1731338258`
换算一下就出来了

最简单就是缓存网页而不是文件

**Reply by refined-fish** [reply]
感谢回复，请问缓存具体是怎么操作呢，我是让机器A定时```curl "订阅连接" -o 配置文件```，然后机器A对外发布```配置文件```这样的。
我尝试浏览器打开订阅链接，查看源代码，没有找到类似```subscription-userinfo: upload=5087212927; download=66988282145; total=536870912000; expire=1731338258```这样的字段，或者说源代码我看起来就是订阅文件本身

**Reply by fejeIaqbhbv** [reply]
你看不到是因为我没说清楚
这玩意叫响应头
源代码形式打开是查看的一种方法的过程之一
浏览器打开源代码就可以在旁边的选项里找到响应头

命令行用curl -i看

**Reply by refined-fish** [reply]
我明白你的意思了，我确实对此了解不多。响应头能静态转存发布吗，是不是需要配置web服务器而没法像文件一样简单的保存发布呀

**Reply by fejeIaqbhbv** [reply]
> 是不是需要配置web服务器

是

---

如果真懒，你可以参考部分机场的做法
把流量信息写进proxy里

**Reply by refined-fish** [reply, marked as answer]
非常感谢回复，去了解了一下，由于不太熟悉nginx配置，而且似乎反代http头和本地文件下载无法共存在一个nginx location中，折腾了半天最终选择了直接使用 proxy_pass 的纯反向代理方式。

虽然我认为更好的方式可能是，服务器A能够定时保存文件并且nginx发布，但客户端在访问服务器A的下载链接时，服务器A又能直接转发订阅连接的http header给客户端（感觉似乎成了在搞CDN）。因为这样即使偶然发生服务器A无法访问订阅链接的情况，也能保证订阅文件能被客户端正确获取到服务器A的上一个备份。

**Comment by grayguest** [comment]
请教楼主是用得什么方式中转的机场订阅信息
